cachi2/core/package_managers/yarn/main.py

```diff
@@ -84,11 +86,15 @@ def _fetch_dependencies(source_dir: RootedPath, output_dir: RootedPath) -> None:
 
     :param source_dir: the directory in which the yarn command will be called.
     :param output_dir: the directory where the yarn dependencies will be downloaded to.
-    :raises CalledProcessError: if the 'yarn install' command fails.
+    :raises YarnBerryError: if the 'yarn install' command fails.
     """
-    # the output_dir is where the globalFolder must be set to
-    # the yarn commands can be called by using the core.utils.run_cmd function
-    pass
+    cachi2_output = output_dir.join_within_root("deps", "yarnberry")
```

Comment:
Any opinions on the file path? It's not _technically_ public API since the output directory structure should be considered opaque, but users will most likely see these paths somewhere, e.g. source containers once RHTAP implements them

If Yarn Berry is _the_ version of Yarn for us, we should probably just call it deps/yarn

Reply:
> If Yarn Berry is the version of Yarn for us, we should probably just call it deps/yarn

Without having literally any knowledge of Yarn, if there are multiple versions/flavours of Yarn packaging you already do support or may support in the future `deps/yarn` will bite us in the back for sure. We should either refer to it by the official name (some lowercase name variant) or yarn + version if those are a thing by which it could be determined whether it's Yarn berry or something else.

Reply:
As for users, if it's not part of the official public API, it doesn't matter at all as the paths are bound to change in that case, so whatever they do on them/expect from them is irrelevant in that case, so we're free to alter it to our liking. Of course the question then is whether we should make these deps paths officially part of the API so we guarantee them - which we should do if we refer to them in any shape or form in the docs... just my 2c

Reply:
There's yarn v1: https://classic.yarnpkg.com/en/docs
And yarn v2+: https://yarnpkg.com/

The official name for both is Yarn, but they might as well be two entirely different package managers. V1 is sometimes referred to as Yarn Classic, v2+ as Yarn Berry.

Yarn v1 is supported in Cachito but not in Cachi2, v2+ is not supported in Cachito and will be in Cachi2. There is _some_ hope that we wouldn't have to implement v1 in Cachi2, but I'm not among the optimists :smile: 

In Cachito Yarn v1 deps went to `deps/yarn`, but I don't think we need to keep backwards compatibility with those, they never mattered to any system or user AFAIK.

Reply:
> Of course the question then is whether we should make these deps paths officially part of the API so we guarantee them - which we should do if we refer to them in any shape or form in the docs... just my 2c

True, we should probably consider changes in the path as breaking changes

Reply:
> I think we should stick with `deps/yarn`. The reasoning is that we'd better stick to a single term throughout the upstream project, either `yarn` or `yarnberry`, and it feels like we are going for the former. It is how it is commonly referred to, AFAICT (simply `yarn`).

I don't disagree but in case the Yarn community decides to go full Google mode and create another fork to supersede yarn berry and make it 'the yarn to go to', then we'd get ourselves into a pickle, especially if we decide to make the mentioned path part of the API. But again, maybe I'm just being overly cautious and thinking too far ahead, ultimately, the decision is a collective one, so if the consensus is that `deps/yarn` is okay, then it's fine by me.

Reply:
Ultimately, for the user, what is the impact of whatever we choose?

* slight confusion over why the file path includes `yarn-with-a-suffix` rather than `yarn` (if I ever look at those)

Anything else?

Reply:
There's a separate matter of what do we call the package manager in the cachi2 input (now this very much _is_ public API)

`cachi2 fetch-deps yarn`?

`cachi2 fetch-deps yarnberry`?

But maybe cachi2 should be smart, accept only `yarn` and auto-detect which yarn it is. We had a story in the backlog to figure this out, but deferred because yarn v1 support may not be needed.

Reply:
TLDR: `deps/yarn` is fine by me :smile:

Reply:
Did not expect such a long discussion about the file path :smile: 

Should I rename YarnBerryError to YarnError to keep it consistent?

Reply:
> Should I rename YarnBerryError to YarnError to keep it consistent?

To me, that makes sense :+1:

Reply:
> Should I rename YarnBerryError to YarnError to keep it consistent?

Yeah, I think dropping Berry from the name makes sense. The error message is probably generic enough to be usable for all flavours of yarn anyway. Maybe call it YarnCommandError though, YarnError seems too generic